src/cpp/lpnamer/main/ProblemGenerationExeOptions.cpp

```diff
@@ -21,19 +23,33 @@ ProblemGenerationExeOptions::ProblemGenerationExeOptions()
                            "use this option if unnamed problems are provided");
 }
 void ProblemGenerationExeOptions::Parse(unsigned int argc,
-                                        const char *const *argv) {
+                                        const char* const* argv) {
   OptionsParser::Parse(argc, argv);
   auto log_location = LOGLOCATION;
-  using namespace std::string_literals;
-  if (!XpansionOutputDir().empty() && !ArchivePath().empty()) {
-    auto msg = "Giving both archive and output options is not permitted"s;
-    throw ProblemGenerationOptions::ConflictingParameters(msg, log_location);
-  }
-  if (XpansionOutputDir().empty() && ArchivePath().empty()) {
-    auto msg = "Both output directory and archive path are empty"s;
+  checkAtLeastOneMandatoryOption(log_location);
+  checkOnlyOneMandatoryOption(log_location);
+}
+void ProblemGenerationExeOptions::checkAtLeastOneMandatoryOption(
+    const std::string& log_location) const {
+  if (std::vector<std::string> mandatory = {this->XpansionOutputDir().string(),
+                                            this->ArchivePath().string(),
+                                            this->StudyPath().string()};
+      std::ranges::all_of(
+          mandatory, [](std::string_view string) { return string.empty(); })) {
+    auto msg =
+        "Need to give at least on of [OutputDir, Archive, Study] options"s;
     throw ProblemGenerationOptions::MissingParameters(msg, log_location);
   }
 }
+void ProblemGenerationExeOptions::checkOnlyOneMandatoryOption(
+    const std::string& log_location) const {
+  if (std::vector<std::string> args = {this->XpansionOutputDir().string(),
+                                       this->ArchivePath().string(), this->StudyPath().string()};
+      std::ranges::count_if(args, std::ranges::empty) < (args.size() - 1)) {
+    auto msg = "Only one of [archive, output, study] parameters is accepted"s;
```

Comment:
- Uniformize error message with the one of the previous function
- Would it be possible to have only one function that checks there is only one and exactly one option specified (as you check at least one and no more than 1) ?

Reply:
Code was simplify according to suggestion